components/esp_diagnostics/include/esp_diagnostics.h

```diff
@@ -126,13 +126,15 @@ typedef struct {
 /**
  * @brief Device information structure
  */
+ #define SHA_SIZE (16 / 2)
+
 typedef struct {
     uint32_t chip_model;                                      /*!< Chip model */
     uint32_t chip_rev;                                        /*!< Chip revision */
     uint32_t reset_reason;                                    /*!< Reset reason */
     char app_version[32];                                     /*!< Application version */
     char project_name[32];                                    /*!< Project name */
-    char app_elf_sha256[CONFIG_APP_RETRIEVE_LEN_ELF_SHA + 1]; /*!< SHA256 of application elf */
+    uint8_t app_elf_sha256[32]; /*!< SHA256 of application elf */
```

Comment:
Can we derive all other numbers from the single `SHA` macro, whatever you choose to define